### PR TITLE
feat(profiler): allow to pass agent URL to the Profiler object

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

File: releasenotes/notes/profiler-url-c61885c7d1fd5ea6.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    It is now possible to pass an `url` parameter to the `Profiler` to specify
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/profiler-url-c61885c7d1fd5ea6.yaml:4]:
```suggestion
    It is now possible to pass a `url` parameter to the `Profiler` to specify
```

File: ddtrace/profiling/profiler.py [hunk 2]
```diff
@@ -129,6 +120,33 @@ def version(self):
...
+
+def _get_default_url(api_key):
+    """Get default profiler exporter URL.
+
+    If an API key is not specified, the URL will default to the agent location configured via the environment variables.
+
+    If an API key is specified, the profiler goes into agentless mode and uses `DD_SITE` to upload directly to Datadog
+    backend.
+    """
+    # Default URL changes if an API_KEY is provided in the env
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/profiler.py:136]: Should we prioritize the url first? I think the UI now nags you to stop using the agentless configuration for profiling. This is slightly annoying if you run the agent in the same environment as the trace client, so DD_API_KEY is defined and used by default by the profiler.

**Review by Kyle-Verhoog — APPROVED** [review]
lgtm, just a question that is out of the scope of this PR

File: ddtrace/profiling/profiler.py (hunk 2 above)
Comment (jd) [reply ddtrace/profiling/profiler.py:136]: The URL is always used if specified (see L180), this is just the default in the user did not specify anything. In that case, we keep doing what we did so far.

Comment (Kyle-Verhoog) [reply ddtrace/profiling/profiler.py:136]: :+1:

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**